devices/nand-imx6ull/bch.c

```diff
@@ -0,0 +1,155 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Operating system loader
+ *
+ * i.MX 6ULL BCH ECC accelerator
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Lukasz Kosinski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include "bch.h"
+
+
+/* BCH registers */
+#define BCH_CTRL          0
+#define BCH_CTRL_SET      1
+#define BCH_CTRL_CLR      2
+#define BCH_CTRL_TOG      3
+#define BCH_STATUS0       4
+#define BCH_STATUS0_SET   5
+#define BCH_STATUS0_CLR   6
+#define BCH_STATUS0_TOG   7
+#define BCH_MODE          8
+#define BCH_MODE_SET      9
+#define BCH_MODE_CLR      10
+#define BCH_MODE_TOG      11
+#define BCH_ENCODEPTR     12
+#define BCH_ENCODEPTR_SET 13
+#define BCH_ENCODEPTR_CLR 14
+#define BCH_ENCODEPTR_TOG 15
+#define BCH_DATAPTR       16
+#define BCH_DATAPTR_SET   17
+#define BCH_DATAPTR_CLR   18
+#define BCH_DATAPTR_TOG   19
+#define BCH_METAPTR       20
+#define BCH_METAPTR_SET   21
+#define BCH_METAPTR_CLR   22
+#define BCH_METAPTR_TOG   23
+#define BCH_LAYOUTSEL     28
+#define BCH_LAYOUTSEL_SET 29
+#define BCH_LAYOUTSEL_CLR 30
+#define BCH_LAYOUTSEL_TOG 31
+#define BCH_LAYOUT0       32
+#define BCH_LAYOUT0_SET   33
+#define BCH_LAYOUT0_CLR   34
+#define BCH_LAYOUT0_TOG   35
+#define BCH_LAYOUT1       36
+#define BCH_LAYOUT1_SET   37
+#define BCH_LAYOUT1_CLR   38
+#define BCH_LAYOUT1_TOG   39
+
+/* BCH NAND layouts */
+#define BCH_LAYOUT_OFFS 8
+
+
+static struct {
+	volatile u32 *const base;
+} bch_common = {
+	.base = (void *)0x1808000,
+};
```

Comment:
Move this to the init, for better or for worse that's our convention

Reply:
Done.